group by

axshai opened the pull request.

axshai committed:
group by

talbarfm commented on 6.5.group_by.py:
```diff
@@ -0,0 +1,13 @@
+    :return: A dictionary where the keys are the results of applying `func` to the elements of `iterable`,
+            and the values are lists of the elements that have that key.
+    """
+    grouped_dict = {func(i): [] for i in iterable}
```
replace 'i' Variable names - https://workat.tech/machine-coding/tutorial/writing-meaningful-variable-names-clean-code-za4m83tiesy0 - variable names, abbreviation reference is missing

axshai committed:
fix group_by

1.replace i with item
2.don't use list comprehension for side effect
3.add run example
